### ISSUE TITLE
Lower Low-Level Accuracy on ChartMimic After Fine-Tuning Qwen2-VL-7B-Instruct with Chart2Code-160k

### ISSUE THREAD

**Issue by JackLingjie** [thunlp/ChartCoder#8]
Hi, thank you for the great work on *ChartCoder: Advancing Multimodal Large Language Model for Chart-to-Code Generation*.

I'm currently trying to reproduce your reported results using **Qwen2-VL-7B-Instruct** fine-tuned on **Chart2Code-160k**, and evaluating on **ChartMimic**. While the **ExecRate** matches your paper, the **Low-Level accuracy** is consistently lower than expected.

Here are the results:

| Model | ExecRate | Low-Level |
|-------|----------|-----------|
| Qwen2-VL-7B (ChartCoder) | 67.00 | 32.90 |
| Qwen2-VL-7B-Instruct + Chart2Code-160k (ChartCoder) | 83.60 | 73.40 |
| Qwen2-VL-7B-Instruct (reproduced) | 69.40 | 34.72 |
| Qwen2-VL-7B-Instruct + Chart2Code-160k (reproduced) | 83.60 | 67.46 |

We used the same model checkpoint (**Qwen2-VL-7B-Instruct**) and successfully reproduced the baseline results, so the setup appears correct.

We followed the training settings described in your paper:

- **Fine-tuning type**: Full finetuning  
- **Batch size**: 128  
- **Learning rate**: 1e-5  
- **Num train epochs**: 1.0 (also tried 3.0, which resulted in slightly worse performance)  
- **LR scheduler**: cosine  
- **Warmup ratio**: 0.03  
- **Max token length**: 16k  

We also tried varying the **temperature** during evaluation, but Low-Level accuracy remains in the **65–67** range.

Could you please share the specific fine-tuning settings you used for **Qwen2-VL-7B-Instruct on Chart2Code-160k**, such as the exact training hyperparameters and other implementation details?

Thank you in advance!

**Comment by xxlllz** [comment]
Hello,

For fine-tuning Qwen2-VL, we follow the settings in [Llama-Factory](https://github.com/hiyouga/LLaMA-Factory/blob/main/examples/train_full/qwen2_5vl_full_sft.yaml). Generally, the settings are
```
Batch size: 32
Warmup ratio: 0.05/0.1
Max token length: 8k (I forget the exact value. It was probably somewhere between 6k and 8k.)
Gradient accumulation steps: 2
```
other settings are same as you provided.

Also, possible reasons for causing this mismatch maybe that our evaluation is performed under the new version of [ChartMimic](https://huggingface.co/datasets/ChartMimic/ChartMimic), which contains 600 samples rather than the 500 samples in the old version. The low-level and high-level score of the model under the new version may increase considerably. You could follow the ChartMimic v1 and v2 versions on arxiv and ChartCoder paper to see this improvement

**Comment by JackLingjie** [comment]
> Hello,
> 
> For fine-tuning Qwen2-VL, we follow the settings in [Llama-Factory](https://github.com/hiyouga/LLaMA-Factory/blob/main/examples/train_full/qwen2_5vl_full_sft.yaml). Generally, the settings are
> 
> ```
> Batch size: 32
> Warmup ratio: 0.05/0.1
> Max token length: 8k (I forget the exact value. It was probably somewhere between 6k and 8k.)
> Gradient accumulation steps: 2
> ```
> 
> other settings are same as you provided.
> 
> Also, possible reasons for causing this mismatch maybe that our evaluation is performed under the new version of [ChartMimic](https://huggingface.co/datasets/ChartMimic/ChartMimic), which contains 600 samples rather than the 500 samples in the old version. The low-level and high-level score of the model under the new version may increase considerably. You could follow the ChartMimic v1 and v2 versions on arxiv and ChartCoder paper to see this improvement

Thank you so much for the quick and detailed response!

We'll try fine-tuning again with the batch size and warmup ratio you mentioned, and also update our evaluation to use the ChartMimic v2 dataset. That may very well explain the difference in Low-Level scores we observed.

Really appreciate your help and clarification — and great work on ChartCoder!

**Comment by MiaSanLei** [comment]
> Hi, thank you for the great work on _ChartCoder: Advancing Multimodal Large Language Model for Chart-to-Code Generation_.
> 
> I'm currently trying to reproduce your reported results using **Qwen2-VL-7B-Instruct** fine-tuned on **Chart2Code-160k**, and evaluating on **ChartMimic**. While the **ExecRate** matches your paper, the **Low-Level accuracy** is consistently lower than expected.
> 
> Here are the results:
> 
> Model	ExecRate	Low-Level
> Qwen2-VL-7B (ChartCoder)	67.00	32.90
> Qwen2-VL-7B-Instruct + Chart2Code-160k (ChartCoder)	83.60	73.40
> Qwen2-VL-7B-Instruct (reproduced)	69.40	34.72
> Qwen2-VL-7B-Instruct + Chart2Code-160k (reproduced)	83.60	67.46
> We used the same model checkpoint (**Qwen2-VL-7B-Instruct**) and successfully reproduced the baseline results, so the setup appears correct.
> 
> We followed the training settings described in your paper:
> 
> * **Fine-tuning type**: Full finetuning
> * **Batch size**: 128
> * **Learning rate**: 1e-5
> * **Num train epochs**: 1.0 (also tried 3.0, which resulted in slightly worse performance)
> * **LR scheduler**: cosine
> * **Warmup ratio**: 0.03
> * **Max token length**: 16k
> 
> We also tried varying the **temperature** during evaluation, but Low-Level accuracy remains in the **65–67** range.
> 
> Could you please share the specific fine-tuning settings you used for **Qwen2-VL-7B-Instruct on Chart2Code-160k**, such as the exact training hyperparameters and other implementation details?
> 
> Thank you in advance!

Hello, I've been trying to reproduce Qwen2-VL-7B's metrics on ChartMimic, but my results differ significantly from those reported in the paper (67.0/32.9/35.0). I'd appreciate if you could share how you reproduced these results and possibly share your code. Thank you very much.

Here are my reproduction results:

Using the code provided in the ChartMimic GitHub repository on direct_600:

| Metric | Exec.Rate | Low-Level | High-Level |
|--------|-----------|-----------|------------|
| Score  | 58.5      | 27.7      | 23.1       |

Using the evaluation code provided by VLMEvalKit on direct_600:

| Metric | Exec.Rate | Low-Level | High-Level |
|--------|-----------|-----------|------------|
| Score  | 61.0      | 29.0      | 28.0       |

Thank you in advance!
